ttnn/cpp/ttnn/operations/transformer/sdpa_decode/device/kernels/dataflow/reader_decode_all.cpp

```diff
@@ -229,6 +246,29 @@ void kernel_main() {
                 noc_async_read_barrier();
                 cb_push_back(cb_k_in, k_chunk_tiles);
 
+                if constexpr(use_attention_mask){
+                    // Read mask chunk
+                    cb_reserve_back(cb_mask_in, mask_chunk_tiles);
+                    uint32_t mask_write_ptr = get_write_ptr(cb_mask_in);
+                    barrier_count = 0;
+                    for (uint32_t row = 0; row < PNHt; ++row) {
+                        uint32_t mask_tile_id = mask_start_tile_id + row * PSt;
+                        for (uint32_t col = 0; col < Sk_chunk_t; ++col) {
+                            noc_async_read_tile(mask_tile_id, mask_reader, mask_write_ptr);
+                            mask_tile_id++;
+                            mask_write_ptr += mask_tile_bytes;
+
+                            if (++barrier_count == barrier_threshold) {
+                                noc_async_read_barrier();
+                                barrier_count = 0;
+                            }
+                        }
```

Comment:
worthwhile to move this into a function? Looks like the same block exists in both branches

Reply:
changed into a function call. Good point!